packages/SwingSet/src/liveslots/virtualReferences.js

```diff
@@ -62,48 +62,69 @@ export function makeVirtualReferenceManager(
     const exportStatus = getExportStatus(vobjID);
     if (exportStatus !== 'reachable' && refCount === 0) {
       let doMoreGC = deleteStoredRepresentation(vobjID);
-      syscall.vatstoreDelete(`vom.rc.${vobjID}`);
-      syscall.vatstoreDelete(`vom.es.${vobjID}`);
-      doMoreGC = doMoreGC || ceaseRecognition(vobjID);
+      const { slot } = parseVatSlot(vobjID);
+      syscall.vatstoreDelete(`vom.rc.${slot}`);
+      syscall.vatstoreDelete(`vom.es.${slot}`);
+      const more = ceaseRecognition(slot);
+      doMoreGC = doMoreGC || more;
       return [doMoreGC, exportStatus !== 'none'];
     }
     return [false, false];
   }
 
   function getRefCount(vobjID) {
-    const raw = syscall.vatstoreGet(`vom.rc.${vobjID}`);
+    const { slot } = parseVatSlot(vobjID);
+    const raw = syscall.vatstoreGet(`vom.rc.${slot}`);
     if (raw) {
       return Number(raw);
     } else {
       return 0;
     }
   }
 
-  function getExportStatus(vobjID) {
-    const raw = syscall.vatstoreGet(`vom.es.${vobjID}`);
-    switch (raw) {
-      case '0':
-        return 'recognizable';
-      case '1':
+  function getExportStatus(slot) {
+    const es = syscall.vatstoreGet(`vom.es.${slot}`);
+    if (es) {
+      if (es.indexOf('r') >= 0) {
         return 'reachable';
-      default:
+      } else if (es.indexOf('s') >= 0) {
+        return 'recognizable';
+      } else {
         return 'none';
+      }
+    } else {
+      return 'none';
     }
   }
 
   function setRefCount(vobjID, refCount) {
-    const { virtual } = parseVatSlot(vobjID);
-    syscall.vatstoreSet(`vom.rc.${vobjID}`, `${Nat(refCount)}`);
+    const { virtual, slot } = parseVatSlot(vobjID);
+    syscall.vatstoreSet(`vom.rc.${slot}`, `${Nat(refCount)}`);
     if (refCount === 0) {
       if (!virtual) {
-        syscall.vatstoreDelete(`vom.rc.${vobjID}`);
+        syscall.vatstoreDelete(`vom.rc.${slot}`);
       }
-      addToPossiblyDeadSet(vobjID);
+      addToPossiblyDeadSet(slot);
+    }
+  }
+
+  function getFacetCount(slot) {
+    const val = requiredValForSlot(slot);
+    if (Array.isArray(val)) {
+      return val.length;
+    } else {
+      return 1;
     }
   }
 
   function setExportStatus(vobjID, exportStatus) {
-    const key = `vom.es.${vobjID}`;
+    const { slot, facet } = parseVatSlot(vobjID);
+    const key = `vom.es.${slot}`;
+    const esRaw = syscall.vatstoreGet(key);
+    const es = Array.from(esRaw || 'n'.repeat(getFacetCount(slot)));
+    const facetIdx = facet === undefined ? 0 : facet;
+    // The export status of each facet is encoded as:
+    // 's' -> 'recogizable' ('s' for "see"), 'r' -> 'reachable', 'n' -> 'none'
```

Comment:
As discussed, I'll make a note to change these flags from `[rsn]` to something with a more obvious ranking, maybe `[Rs_]` (where the vertical height of the letter indicates how "strong" the reference is), or maybe `[Rc_]` because there's a `c` in `reCognizable`. Using `[210]` would be the most obvious, but you made an excellent point that it would be horribly confused with a multi-digit integer reference count.

I won't implement that change until I have some free time, which means possibly never.